internal/cloud/cloud.go

```diff
@@ -101,6 +97,28 @@ func Shell(configDir string) error {
 	return shell(username, vmHostname, configDir)
 }
 
+func enactGithubUsernameStep() string {
+	stepGithubUsername := stepper.Start("Detecting your Github username...")
+
+	usernameSource, username, err := auth.Username()
+	if err != nil || usernameSource != auth.GithubSource {
+		if err != nil {
+			debug.Log("failed to get auth.Username. Error: %v", err)
+		}
+
+		username, err = queryGithubUsername()
+		if err == nil && username != "" {
+			stepGithubUsername.Success("Username: %s", username)
+		} else {
+			stepGithubUsername.Fail("Unable to resolve username")
+			// The query for Github username is best effort, and if it fails to resolve
+			// we fallback to prompting the user, and suggesting the local computer username.
+			username = promptUsername()
```

Comment:
This might be something for a different PR, but it'd be nice to also try the current username. The flow would be something like:

1. Query GitHub for username. If that fails, default to the current username.
2. Try to authenticate. If that fails, prompt for the username and try again.
3. Upon success save the GitHub username.

Reply:
We could try that...